server/zio-http/src/main/scala/sttp/tapir/server/zhttp/ZHttpInterpreter.scala

```diff
@@ -0,0 +1,140 @@
+package sttp.tapir.server.zhttp
+
+import io.netty.handler.codec.http.HttpResponseStatus
+import sttp.capabilities
+import sttp.capabilities.zio.ZioStreams
+import sttp.model.{HasHeaders, QueryParams, Uri, Header => SttpHeader, Method => SttpMethod}
+import sttp.monad.MonadError
+import sttp.tapir.model.{ConnectionInfo, ServerRequest}
+import sttp.tapir.server.interpreter.{RequestBody, ServerInterpreter, ToResponseBody}
+import sttp.tapir.{CodecFormat, Endpoint, RawBodyType, RawPart, WebSocketBodyOutput}
+import zhttp.http.{CanSupportPartial, Http, HttpChannel, HttpContent, Request, Response, Status, Header => ZHttpHeader}
+import zio._
+import zio.stream._
+
+import java.net.InetSocketAddress
+import java.nio.charset.{Charset, StandardCharsets}
+import scala.collection.immutable.Seq
+
+class ZHttpServerRequest(req: Request) extends ServerRequest {
+  def protocol: String = "HTTP/1.1" //TODO
+
+  def local =
+    for {
+      host <- req.url.host
+      port <- req.url.port
+    } yield new InetSocketAddress(host, port)
+
+  lazy val connectionInfo: ConnectionInfo =
+    ConnectionInfo(local, None, None)
+
+  def underlying: Any = req
+
+  lazy val pathSegments: List[String] = req.url.path.toList
+  lazy val queryParameters: QueryParams = QueryParams.fromMap(Map.empty) // TODO: parse
+
+  def method: SttpMethod = SttpMethod(req.method.asJHttpMethod.name().toUpperCase)
+
+  def uri: Uri = Uri.unsafeParse(req.url.toString()) //TODO: is this correct?
+
+  lazy val headers: Seq[SttpHeader] = req.headers.map(h => SttpHeader(h.name.toString, h.value.toString))
+}
+
+class ZHttpRequestBody[R](request: Request) extends RequestBody[RIO[R, *], ZioStreams] {
+  override val streams: capabilities.Streams[ZioStreams] = ZioStreams
+
+  def toRaw[R](bodyType: RawBodyType[R]): Task[R] = bodyType match {
+    case RawBodyType.StringBody(_) =>
+      request.data.content match {
+        case HttpContent.Complete(data) => Task.succeed(data)
+        case HttpContent.Chunked(data) => data.fold("")(_ + _)
+      }
+    case RawBodyType.ByteArrayBody =>
+      request.data.content match {
+        case HttpContent.Complete(data) => Task.succeed(data.getBytes())
+        case HttpContent.Chunked(data) => data.fold("")(_ + _).map(_.getBytes)
```

Comment:
we shouldn't go through a `String` to get the bytes - as conversions (due to charsets etc) might malform the data. Maybe there's a way to concatenate byte chunks?

Reply:
Yes, this is wrong. Will correct once we have a Stream of bytes :)